osu.Game.Rulesets.Taiko/Mods/TaikoModClassic.cs

```diff
@@ -3,21 +3,200 @@
 
 #nullable disable
 
+using System.Linq;
+using System.Threading;
+using osu.Game.Beatmaps;
+using osu.Game.Rulesets.Judgements;
 using osu.Game.Rulesets.Mods;
+using osu.Game.Rulesets.Scoring;
+using osu.Game.Rulesets.Taiko.Beatmaps;
+using osu.Game.Rulesets.Taiko.Judgements;
 using osu.Game.Rulesets.Taiko.Objects;
+using osu.Game.Rulesets.Taiko.Objects.Drawables;
 using osu.Game.Rulesets.Taiko.UI;
 using osu.Game.Rulesets.UI;
 
 namespace osu.Game.Rulesets.Taiko.Mods
 {
-    public class TaikoModClassic : ModClassic, IApplicableToDrawableRuleset<TaikoHitObject>, IUpdatableByPlayfield
+    public class TaikoModClassic : ModClassic, IApplicableToDrawableRuleset<TaikoHitObject>, IUpdatableByPlayfield, IApplicableAfterBeatmapConversion
     {
         private DrawableTaikoRuleset drawableTaikoRuleset;
 
         public void ApplyToDrawableRuleset(DrawableRuleset<TaikoHitObject> drawableRuleset)
         {
             drawableTaikoRuleset = (DrawableTaikoRuleset)drawableRuleset;
             drawableTaikoRuleset.LockPlayfieldAspect.Value = false;
+
+            drawableTaikoRuleset.Playfield.RegisterPool<ClassicDrumRoll, ClassicDrawableDrumRoll>(5);
+            drawableTaikoRuleset.Playfield.RegisterPool<ClassicDrumRollTick, DrawableDrumRollTick>(100);
+            drawableTaikoRuleset.Playfield.RegisterPool<ClassicSwell, ClassicDrawableSwell>(5);
+        }
+
+        public void ApplyToBeatmap(IBeatmap beatmap)
+        {
+            var taikoBeatmap = (TaikoBeatmap)beatmap;
+
+            if (taikoBeatmap.HitObjects.Count == 0) return;
+
+            var hitObjects = taikoBeatmap.HitObjects.Select(ho =>
+            {
+                if (ho is DrumRoll drumRoll)
+                {
+                    var newDrumRoll = new ClassicDrumRoll(drumRoll);
+                    return newDrumRoll;
+                }
+
+                if (ho is Swell swell)
+                {
+                    var newSwell = new ClassicSwell(swell);
+                    return newSwell;
+                }
+
+                return ho;
+            }).ToList();
+
+            taikoBeatmap.HitObjects = hitObjects;
+        }
+
+        private class ClassicDrumRoll : DrumRoll
+        {
+            public ClassicDrumRoll(DrumRoll original)
+            {
+                StartTime = original.StartTime;
+                Samples = original.Samples;
+                EndTime = original.EndTime;
+                Duration = original.Duration;
+                TickRate = original.TickRate;
+                RequiredGoodHits = original.RequiredGoodHits;
+                RequiredGreatHits = original.RequiredGreatHits;
+            }
+
+            public override Judgement CreateJudgement() => new TaikoClassicDrumRollJudgement();
+
+            protected override void CreateTicks(CancellationToken cancellationToken)
```

Comment:
I would prefer this method to not re-implement the base method letter-by-letter, but rather do something like the following:

```csharp
protected override void CreateTicks(CancellationToken cancellationToken)
{
    var ticks = base.CreateTicks(cancellationToken);
    // map DrumRollTicks to ClassicDrumRollTicks
}
```

The major obstacle to this is the `AddNested()` call, but you could change the signature to return an enumerable instead, and only call `AddNested()` in the base `CreateNestedHitObjects()`.